garg2021re/openreview/sections/methodology.tex

```diff
@@ -0,0 +1,42 @@
+\section{Methodology}
+
+We have re-implemented the algorithm proposed in the paper from scratch using PyTorch and created an end-to-end model trainer with a Keras-like interface. We referred to the code given by the authors for the baseline model hyperparameters and the source of synthetic datasets. The algorithm presented by the original authors in their paper can be represented as follows:
+
+\begin{algorithm}[H]
+\SetAlgoLined
+\DontPrintSemicolon
+
+\kwInput{\textbf{Input:} Initialize $\lambda$} \;
```

Comment:
- Instead of using `\kwInput`, define a macro using `SetKwInOut`
- Also, fix the dangling `\tanh` within the `$..$` expression
- In this tex, remove `\limit` which is undefined. Use just `\sum` or `\lim` instead (https://www.overleaf.com/learn/latex/Integrals,_sums_and_limits)

```
\begin{algorithm}[H]
\SetAlgoLined
\DontPrintSemicolon
\SetKwInOut{Input}{Input}

\Input{Initialize $\lambda$} \;
\For{number of training epochs}{
    \For{ i = 1,...,number of mini-batch examples}{
        Sample $ \epsilon \sim \mathcal{U}(0,1) $ and set $\delta = \frac{1}{2}\log\frac{\epsilon}{1-\epsilon}$ \;
        Initialize $w_{b}$ = $\tanh((\lambda + \delta)/\tau$) \;
        Compute following using \textit{gumbel-softmax} trick \[ g_i :=  \frac{1}{M} \nabla_{w_{b}}l(y_{i}, f_{w_{r}}(x_{i})) \] \[s_i := \frac{N(1-w^2_{b})}{\tau(1-\tanh(\lambda)^2)}\]
    }
    Update $\mu$ and $\lambda$ using following equation \[\mu \leftarrow \tanh(\lambda) \] \[\lambda \leftarrow (1-\alpha)\lambda - \alpha[\sum_{i=1}^{M} (s_i\odot g_i) - \lambda_{0}]\]
}
 
 \caption{Bayesian Learning rule for BayesBiNN}
\label{alg:alg1}
\end{algorithm}
```